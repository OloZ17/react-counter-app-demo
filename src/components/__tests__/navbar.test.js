import React from 'react';
import { render, screen } from '@testing-library/react';
import NavBar from '../navbar';

describe('Testing Navbar() Component', () => {
  const totalCounters = 10;

  it('should take a snapshot', () => {
    const { asFragment } = render(<NavBar totalCounters={totalCounters} />);
    expect(asFragment(<NavBar />)).toMatchSnapshot();
  });

  it('should render totalCounters', () => {
    render(<NavBar totalCounters={totalCounters} />);
    expect(screen.getByText(`${totalCounters}`)).toBeInTheDocument();
    expect(screen.getByText(10)).toBeInTheDocument();
  });
});
