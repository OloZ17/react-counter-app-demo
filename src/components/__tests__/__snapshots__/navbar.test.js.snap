// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar() Component should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-light"
  >
    <div
      class="navbar-brand"
    >
      <i
        aria-hidden="true"
        class="fa fa-shopping-cart fa-lg m-2"
      />
      <span
        class="badge badge-pill bg-info m-2"
        style="width: 50px; font-size: 24px;"
      >
        10
      </span>
      Items
    </div>
  </nav>
</DocumentFragment>
`;
